packages/core/test/specs/data_sources/model/collection_component/CollectionComponent.ts

```diff
@@ -0,0 +1,863 @@
+      ],
+    });
+
+    firstRecord = dataSource.getRecord('user1')!;
+    secondRecord = dataSource.getRecord('user2')!;
+  });
+
+  afterEach(() => {
+    em.destroy();
+  });
+
+  test('Collection component should be undroppable', () => {
+    const cmp = wrapper.components({
+      type: CollectionComponentType,
+      collectionDefinition: {
+        block: {
+          type: 'default',
+        },
+        config: {
+          dataSource: {
+            type: DataVariableType,
+            path: 'my_data_source_id',
+          },
+        },
+      },
+    })[0];
+
+    expect(cmp.get('droppable')).toBe(false);
+  });
+
+  test('Collection items should be undraggable', () => {
+    const cmp = wrapper.components({
+      type: CollectionComponentType,
+      collectionDefinition: {
+        block: {
+          type: 'default',
+        },
+        config: {
+          dataSource: {
+            type: DataVariableType,
+            path: 'my_data_source_id',
+          },
+        },
+      },
+    })[0];
+
+    cmp.components().forEach((child) => {
+      expect(child.get('draggable')).toBe(false);
+    });
+  });
+
+  test('Collection items should be symbols', () => {
+    const cmp = wrapper.components({
+      type: CollectionComponentType,
+      collectionDefinition: {
+        block: {
+          type: 'default',
+          components: [
+            {
+              type: 'default',
+            },
+          ],
+        },
+        config: {
+          dataSource: {
+            type: DataVariableType,
+            path: 'my_data_source_id',
+          },
+        },
+      },
+    })[0];
+
+    expect(cmp.components()).toHaveLength(3);
+    cmp.components().forEach((child) => expect(child.get('type')).toBe('default'));
+    const children = cmp.components();
+    const firstChild = children.at(0);
+
+    children.slice(1).forEach((component) => {
+      expect(getSymbolMain(component)).toBe(firstChild);
+    });
+  });
+
+  describe('Collection variables', () => {
+    describe('Properties', () => {
+      let cmp: Component;
+      let firstChild!: Component;
+      let firstGrandchild!: Component;
+      let secondChild!: Component;
+      let secondGrandchild!: Component;
+      let thirdChild!: Component;
+
+      beforeEach(() => {
+        cmp = wrapper.components({
+          type: CollectionComponentType,
+          collectionDefinition: {
+            block: {
+              type: 'default',
+              components: [
+                {
+                  type: 'default',
+                  content: {
+                    type: CollectionVariableType,
+                    variable_type: CollectionStateVariableType.current_item,
+                    path: 'user',
+                  },
+                },
+              ],
+              content: {
+                type: CollectionVariableType,
+                variable_type: CollectionStateVariableType.current_item,
+                path: 'user',
+              },
```

Comment:
Let's avoid using `componentDef.content`, it's a bit of edge–case thing (custom RTE). 
The goal of `content` is to get/render the HTML as it is, without parsing it to components. So it doesn't even make sense to have a component with both `componentDef.content` and `componentDef.components`.